### PR TITLE
Add +pretty-code/install-iosevka-font

### PR DESCRIPTION
**Pull request by jsravn**
In response to https://github.com/hlissner/doom-emacs/issues/797#issuecomment-546027790.

This adds a command `+pretty-code/install-iosevka-font` when `(pretty-code +iosevka)` is enabled. This downloads the Iosevka font with support for ligatures in Emacs that was built by @ar1a.

I wasn't sure where to put the fonts - for now I've put them at https://github.com/jsravn/iosevka-emacs which can be moved elsewhere.

The implementation is just a lightly modified version of all-the-icons.el install-fonts command.

### CODE REVIEW

**Comment by Vvkmnn** [comment]
Seems like `doom-path` is missing from the CI build.

**Comment by hlissner** [comment]
@Vvkmnn It was fixed in a later commit.

**Comment by MarioRicalde** [comment]
What is pending to get this merged?

File: modules/ui/pretty-code/+iosevka.el [hunk 1]
```diff
@@ -230,3 +230,54 @@
...
+
+      (dolist (font +pretty-code--iosevka-font-names)
+        (url-copy-file (format url-format font) (expand-file-name font font-dest) t))
+
+      (when known-dest?
+        (message "Font downloaded, updating font cache... <fc-cache -f -v> ")
+        (shell-command-to-string (format "fc-cache -f -v")))
+      (message "Successfully %s `Iosevka' fonts to `%s'!"
+               (if known-dest? "installed" "downloaded")
+               font-dest))))
```
Comment (hlissner) [inline modules/ui/pretty-code/+iosevka.el]: Please move this to `ui/pretty-code/autoload/iosevka.el`? (and move `ui/pretty-code/autoload.el` to `ui/pretty-code/autoload/pretty-code.el`)

**Comment by kayg04** [comment]
A semi-bold variant of "Iosevka" would be really useful as the original font does include such a variant.

**Comment by shonfeder** [comment]
Oh man! So close to having this in place already :) I don't have bandwidth to help this along at the moment, but would like to see support for fira code.

I'm commenting here as a ping, and so I'll follow, and I'll stick a TODO in my org file to help address the change requests, and we'll see if I ever get to that item :)

**Comment by jsravn** [comment]
I'm planning to do this, just been swamped with other work.

**Comment by jsravn** [comment]
@hlissner I think your review comments are addressed now. Programata Pro is a commercial font so I couldn't add an auto-download command for it.

**Comment by jsravn** [comment]
@hlissner checking you didn't miss this. :)

File: modules/ui/pretty-code/autoload/fira.el [hunk 2]
```diff
@@ -0,0 +1,20 @@
+;;; ui/pretty-code/autoload/fira.el -*- lexical-binding: t; -*-
+
+(defvar +pretty-code--fira-font-names
+  '(
+    "FiraCode-Bold.ttf"
```
Comment (ar1a) [inline modules/ui/pretty-code/autoload/fira.el]: Should this be on the same line?

File: modules/ui/pretty-code/autoload/hasklig.el [hunk 3]
```diff
@@ -0,0 +1,29 @@
+;;; ui/pretty-code/autoload/hasklig.el -*- lexical-binding: t; -*-
+
+(defvar +pretty-code--hasklig-font-names
+  '(
+    "Hasklig-Black.otf"
```
Comment (ar1a) [inline modules/ui/pretty-code/autoload/hasklig.el]: Same here

File: modules/ui/pretty-code/autoload/iosevka.el [hunk 4]
```diff
@@ -0,0 +1,36 @@
+;;; ui/pretty-code/autoload/iosevka.el -*- lexical-binding: t; -*-
+
+(defvar +pretty-code--iosevka-font-names
+  '(
+    "iosevka-custom-lightoblique.ttf"
```
Comment (ar1a) [inline modules/ui/pretty-code/autoload/iosevka.el]: And here

**Comment by hlissner** [comment]
Sorry for the late response. This looks good to me. Thanks for the help!

**Comment by shonfeder** [comment]
@jsravn I had occasion to use this today, and it was incredibly helpful. Thank you so much for your work!

**Comment by jsravn** [comment]
@shonfeder glad it worked for you!

**Pull request merged**